src/plugins/intel_cpu/tests/functional/single_layer_tests/nms_rotated.cpp

```diff
@@ -0,0 +1,338 @@
+        configuration               = std::get<16>(params);
+
+        std::tie(inFmts, outFmts, priority, selectedType) = cpu_params;
+
+        std::vector<InputShape> actual_shapes;
+        ov::ParameterVector in_params;
+        std::vector<std::shared_ptr<ov::Node>> inputs;
+        const auto in_shape_1d = InputShape{{1}, {{1}}};
+
+#define CONST_CASE(P, S, H, L)                                                                                                 \
+        case P:                                                                                                                \
+            inputs.push_back(ngraph::builder::makeConstant(P, S, std::vector<ov::element_type_traits<P>::value_type>{}, true,  \
+                             ov::element_type_traits<P>::value_type(H), ov::element_type_traits<P>::value_type(L)));           \
+            break;
+
+#define CREATE_INPUT(C, P, S, N, H, L)                                                                                         \
+        if (C) {                                                                                                               \
+            switch (P) {                                                                                                       \
+                CONST_CASE(ElementType::f32,  S.second[0], H, L)                                                               \
+                CONST_CASE(ElementType::f16,  S.second[0], H, L)                                                               \
+                CONST_CASE(ElementType::bf16, S.second[0], H, L)                                                               \
+                CONST_CASE(ElementType::i32,  S.second[0], H, L)                                                               \
+                CONST_CASE(ElementType::i64,  S.second[0], H, L)                                                               \
+            }                                                                                                                  \
+        } else {                                                                                                               \
+            actual_shapes.push_back(S);                                                                                        \
+            if (S.first.rank() == 0) {                                                                                         \
+                in_params.push_back(std::make_shared<ov::op::v0::Parameter>(P, S.second.front()));                             \
+            } else {                                                                                                           \
+                in_params.push_back(std::make_shared<ov::op::v0::Parameter>(P, S.first));                                      \
+            }                                                                                                                  \
```

Comment:
If we call ` init_input_shapes(actual_shapes);` before inputs creation, we can just pass `inputDynamicShapes[i]` to this macros and avoid this `if`. I don't insist though
P.S. However, we need to fill `actual_shapes` before input shapes initialization to follow my suggestion.

Reply:
There are three entities here that depend on each other and on the order of constants: `actual_shapes`, `inputs` and `in_params`, thus better to fill all of them first then call fn `init_input_shapes`.